homeassistant/components/tesla_fleet/climate.py

```diff
@@ -0,0 +1,330 @@
+    """Tesla Fleet vehicle climate entity."""
+
+    _attr_precision = PRECISION_HALVES
+
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_hvac_modes = [HVACMode.HEAT_COOL, HVACMode.OFF]
+    _attr_supported_features = (
+        ClimateEntityFeature.TURN_ON
+        | ClimateEntityFeature.TURN_OFF
+        | ClimateEntityFeature.TARGET_TEMPERATURE
+        | ClimateEntityFeature.PRESET_MODE
+    )
+    _attr_preset_modes = ["off", "keep", "dog", "camp"]
+    _enable_turn_on_off_backwards_compatibility = False
+
+    def __init__(
+        self,
+        data: TeslaFleetVehicleData,
+        side: TeslaFleetClimateSide,
+        scopes: Scope,
+    ) -> None:
+        """Initialize the climate."""
+
+        self.read_only = Scope.VEHICLE_CMDS not in scopes or data.signing
+
+        if self.read_only:
+            self._attr_supported_features = ClimateEntityFeature(0)
+            self._attr_hvac_modes = []
+
+        super().__init__(
+            data,
+            side,
+        )
+
+    def _async_update_attrs(self) -> None:
+        """Update the attributes of the entity."""
+        value = self.get("climate_state_is_climate_on")
+        if value is None:
+            self._attr_hvac_mode = None
+        elif value:
+            self._attr_hvac_mode = HVACMode.HEAT_COOL
+        else:
+            self._attr_hvac_mode = HVACMode.OFF
+
+        # If not scoped, prevent the user from changing the HVAC mode by making it the only option
+        if self._attr_hvac_mode and self.read_only:
+            self._attr_hvac_modes = [self._attr_hvac_mode]
+
+        self._attr_current_temperature = self.get("climate_state_inside_temp")
+        self._attr_target_temperature = self.get(f"climate_state_{self.key}_setting")
+        self._attr_preset_mode = self.get("climate_state_climate_keeper_mode")
+        self._attr_min_temp = cast(
+            float, self.get("climate_state_min_avail_temp", DEFAULT_MIN_TEMP)
+        )
+        self._attr_max_temp = cast(
+            float, self.get("climate_state_max_avail_temp", DEFAULT_MAX_TEMP)
+        )
+
+    async def async_turn_on(self) -> None:
+        """Set the climate state to on."""
+
+        await self.wake_up_if_asleep()
+        await handle_vehicle_command(self.api.auto_conditioning_start())
+
+        self._attr_hvac_mode = HVACMode.HEAT_COOL
+        self.async_write_ha_state()
+
+    async def async_turn_off(self) -> None:
+        """Set the climate state to off."""
+
+        await self.wake_up_if_asleep()
+        await handle_vehicle_command(self.api.auto_conditioning_stop())
+
+        self._attr_hvac_mode = HVACMode.OFF
+        self._attr_preset_mode = self._attr_preset_modes[0]
+        self.async_write_ha_state()
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set the climate temperature."""
+
+        if ATTR_TEMPERATURE not in kwargs:
+            raise ServiceValidationError(
```

Comment:
This line isn't covered by tests but it seems like the intents of the tests were to cover this.  Maybe its because it's only covered by one of the entity types? consider making a parameterized test with both types.

Reply:
I've added a new test to cover this specifically.